components/proxy/src/test/kotlin/com/hotels/styx/services/OriginsAdminHandlerTest.kt

```diff
@@ -167,10 +171,21 @@ class OriginsAdminHandlerTest : FeatureSpec({
             expectStateChange(STATE_UNREACHABLE, STATE_ACTIVE, STATE_UNREACHABLE, false)
         }
 
-        scenario("Activating an inactive origin results in an active state") {
+        // TODO: Complete this test
+        scenario("Activating an unhealthchecked, inactive origin results in an active state") {
             expectStateChange(STATE_INACTIVE, STATE_ACTIVE, STATE_ACTIVE, false)
         }
 
+        // TODO: Complete this test
+        scenario("Activating a healthchecked, inactive origin results in an unreachable state") {
+            expectStateChange(STATE_INACTIVE, STATE_ACTIVE, STATE_UNREACHABLE, false, "app.origin.hc.running")
+        }
+
+        // TODO: Complete this test
+        scenario("Activating a healthchecked (but the healthchecker is stopped), inactive origin results in an unreachable state") {
+            expectStateChange(STATE_INACTIVE, STATE_ACTIVE, STATE_ACTIVE, false, "app.origin.hc.stopped")
+        }
```

Comment:
Please remind me, when do we end up stopping a health checker (if that is active)?

Reply:
In practice, only when the origins config is updated to remove healthchecking configuration for an origin altogether. Which does mean that this case is unlikely. But I was bothered by the fact that the service does support being stopped, and if it was, then there would be nothing to move the state of the origin from `unreachable` to `active`. So, straying into YAGNI territory a bit, I guess. Happy to excise this if you reckon it's not needed.

Reply:
> So, straying into YAGNI territory a bit, I guess. Happy to excise this if you reckon it's not needed.

No this is an important consideration. Just that this scenario didn't immediately occur to me.

Based on your response, I understand there are two possible scenarios:

1) Health check configuration is removed from origins configuration.
2) Origins configuration remains unchanged, but somebody stops the service over the admin interface?

Case 1: Removal of health check configuration only:
  - Looking at code, in this case the host proxy is re-tagged with STATE_ACTIVE. Which seems correct. 
  - Looking at `updateRoutingObjects` in `YamlFileConfigurationService`, I do see a possible bug. The update function doesn't take the tags into account. Suppose the tags alone change, but object's YAML configuration remains unchanged. This code doesn't seem to update the underlying routing object record in this case: WDYT?

```
     objectDefs.forEach { objectDef ->
            routeDb.compute(objectDef.name()) { previous ->
                if (previous == null || changed(objectDef.config(), previous.config)) {
                    // Possible optimisation: check if the tags alone changed, or if the configuration changed too.
                    previous?.routingObject?.stop()
                    converter.routingObjectRecord(objectDef)
                } else {
                    previous
                }
            }
        }
```

Case 2: Looking at code, it removes the health check tag, and sets `state=active`.

Are there any other scenarios we'd have to take into account?

Reply:
> The update function doesn't take the tags into account. 
One side-effect of doing so is that, even if you haven't actually changed the tags in the config file, but the tags have changed at runtime, then you will reset the tags to the configured set each time you reload the origins file. So if you've deactivated an origin via the admin interface, you will reactivate it by reloading the origins file, even if you didn't change the tag in the origins file.

Reply:
The current implementation also resets the tags, but only if there has been a config change for that origin. Which seems a bit more reasonable, even if not perfect.